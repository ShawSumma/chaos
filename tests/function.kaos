void func a()
    print "inside function a"
    num val1 = 5
    print val1
end

print "first print this"

a()
b()

null function b()
    print "inside function b"
    num val2 = 8
    print val2
end

str hello = 'hello world'
number pi = 3.14

str param1 = 'testing'

nil def f1(bool param1, num param2, str param3)
    param3 = 'hello my friend'
	print param1
	print param2
	print param3
    symbol_table
end

f1(true, 5, hello)
f1(false, pi, "OMG!")

print hello
print pi

array c = [1, 2, 3]

array def f2(string param1, array param2)
    print param1
    return param2
end

print f2('Array function parameter:', c)

dict d = {'a': 'foo', 'b': 'bar', 'c': 'baz'}

dict func f3(string param1, dict param2)
    print param1
    return param2
end

print f3('Dictionary function parameter:', d)

num def add(num x, num y)
    num result = x + y
    return result
end

print add(3, 5)
