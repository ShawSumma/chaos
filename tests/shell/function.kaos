nil def f1()
    print "hey"
end

asdasd1()

f1()

asdasd2()

num def f2()
    num a = 5
    return a
end
f2()
