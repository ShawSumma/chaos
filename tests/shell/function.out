Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 7, in <module>
asdasd1()
Undefined function: asdasd1
Absorbed by Interactive Shell
hey
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 11, in <module>
asdasd2()
Undefined function: asdasd2
Absorbed by Interactive Shell
5
4
101
101
